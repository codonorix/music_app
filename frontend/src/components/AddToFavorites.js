import axios from 'axios';
import moment from "moment";

// Add
export default function AddToFavorites(props) {
    axios.post('/favorites', {
        "trackName": props.trackName,
        "kind": props.kind ? props.kind : props.wrapperType,
        "artistName": props.artistName,
        "previewUrl": props.previewUrl,
        "id": props.trackId ? props.trackId : props.collectionId,
        "genre": props.primaryGenreName ? props.primaryGenreName : null,
        "releaseDate": moment(props.releaseDate).utc().format('YYYY'),
        "artworkUrl": props.artworkUrl100,

    })
        .then(function (response) {
            alert("Item added to favorites!");
        })
        .catch(function (error) {
            console.log(error);
        });
}