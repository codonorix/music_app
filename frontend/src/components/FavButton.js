import React from 'react';
import {Link} from "react-router-dom";

export const FavButton = () => {
    return (
        <button className={'btn btn-primary mb-2'}><Link className={'favBtn'} to={'/favorites'}>My Favorites</Link>
        </button>
    )
}
