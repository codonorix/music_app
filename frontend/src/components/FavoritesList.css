.card {
    text-align: center;
}

.favBtn {
    color: white;
}

.video {
    margin: auto;
    max-width: 100%;
}