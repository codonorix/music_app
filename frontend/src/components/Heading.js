import React from 'react';

export const Heading = () => {
    return (
        <div>
            <h2>My Media List</h2>
            <h4>Search for your favorite music, music videos, podcasts, and more!</h4>
            <h4 className={'mb-4'}>Add them to a list of your favorite media items.</h4>
        </div>
    )
}