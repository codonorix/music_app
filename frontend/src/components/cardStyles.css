.card {
    width: 90%;
    margin: auto;
    background-color: rgba(161, 208, 191, 0.69);
    box-shadow: 2px 3px 3px #abbebe;
}

.card-img-top {
    max-width: 100px;
    max-height: 100px;
    margin: auto;
}

.video {
    margin: auto;
    max-width: 100%;
}