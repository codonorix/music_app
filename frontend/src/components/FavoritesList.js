import React, {useState} from "react";
import {useEffect} from "react";
import {Link} from "react-router-dom";
import './FavoritesList.css'
import RemoveFavorites from './RemoveFavorites'

export default function FavoritesList() {

    // Set states
    const [allFavs, setAllFavs] = useState([])

    //API Call to get json file
    useEffect(() => {
        fetch('/favorites')
            .then(res => res.json())
            .then(res => {
                setAllFavs(res)
            })
            .catch(err => {
                console.log(err)
            })
    }, [allFavs])

    return (
        // Map through allFavs to display on page
        <div>
            <button className={'btn btn-primary'}><Link to={'/'} className={'favBtn'}>Back to search</Link></button>
            <h2 className={'text-center'}>My Favorites</h2>
            <div className="row row-cols-1 row-cols-md-3 g-4 mt-2">
                {allFavs.length >= 1 ? allFavs.map((item) => (
                        <div className={'col'} key={item.id}>
                            <div className="card h-100">
                                <img src={item.artworkUrl} className="card-img-top" alt="cover image"/>
                                <div className="card-body">
                                    <small
                                        className={'fst-italic'}>{item.kind.charAt(0).toUpperCase() + item.kind.slice(1)} • {item.genre} • {item.releaseDate}</small>
                                    <h4 className="card-title pt-1">{item.artistName}</h4>
                                    <p className="card-text">{item.trackName}</p>
                                    {item.kind === 'song' ? <audio controls>
                                        <source src={item.previewUrl} type="audio/mpeg"/>
                                    </audio> : null}
                                    {item.kind === 'feature-movie' || item.kind === 'music-video' || item.kind === 'tv-episode' ?
                                        <video className={'video'} controls>
                                            <source src={item.previewUrl} type="video/mp4"/>
                                        </video> : null}
                                </div>
                                <small>{item.id}</small>
                                <div className="card-footer">
                                    <button className={'btn btn-secondary'} onClick={() => RemoveFavorites(item)}>Remove
                                        from Favorites
                                    </button>
                                </div>
                            </div>
                        </div>
                    )
                ) : <p>You have not added any items to your favorites!</p>}
            </div>
        </div>
    )
}