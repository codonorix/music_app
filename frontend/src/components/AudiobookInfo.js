import React from 'react';
import moment from "moment";
import './cardStyles.css'
import AddToFavorites from "./AddToFavorites";

export default function AudiobookInfo(props) {
    return (
        <div className={'col'}>
            <div className="card h-100">
                <img src={props.artworkUrl100} className="card-img-top" alt="cover image"/>
                <div className="card-body">
                    <small className={'fst-italic'}>{props.wrapperType.charAt(0).toUpperCase() + props.wrapperType.slice(1)} â€¢ {moment(props.releaseDate).utc().format('YYYY')}</small>
                    <h5 className="card-title">{props.artistName}</h5>
                    <p className="card-text">{props.trackName}</p>
                </div>
                <small>{props.collectionId}</small>
                <div className="card-footer">
                    <button className={'btn btn-secondary'} onClick={() => AddToFavorites(props)}>Add to favorites</button>
                </div>
            </div>
        </div>
    )
}