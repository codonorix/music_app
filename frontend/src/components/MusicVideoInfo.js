import React from 'react';
import moment from "moment";
import './cardStyles.css'
import AddToFavorites from "./AddToFavorites";

export default function MusicVideoInfo(props) {
    return (
        <div className={'col'}>
            <div className="card h-100">
                <img src={props.artworkUrl100} className="card-img-top" alt="cover image"/>
                <div className="card-body">
                    <small className={'fst-italic'}>{props.kind.charAt(0).toUpperCase() + props.kind.slice(1)} â€¢ {moment(props.releaseDate).utc().format('YYYY')}</small>
                    <h5 className="card-title pt-1">{props.artistName}</h5>
                    <p className="card-text">{props.trackName}</p>
                </div>
                <video className={'video'} controls>
                    <source src={props.previewUrl} type="video/mp4"/>
                </video>
                <small>{props.trackId}</small>
                <div className="card-footer">
                    <button className={'btn btn-secondary'} onClick={() => AddToFavorites(props)}>Add to favorites</button>
                </div>
            </div>
        </div>
    )
}