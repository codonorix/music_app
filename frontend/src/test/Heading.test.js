import React from 'react';
import renderer from 'react-test-renderer';
import {Heading} from "../components/Heading";

test('snapshot test renders correctly', () => {
    const testRenderer = renderer.create(<Heading />);
    expect(testRenderer.toJSON()).toMatchSnapshot()
})