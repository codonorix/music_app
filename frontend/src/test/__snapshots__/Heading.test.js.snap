// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test renders correctly 1`] = `
<div>
  <h2>
    My Media List
  </h2>
  <h4>
    Search for your favorite music, music videos, podcasts, and more!
  </h4>
  <h4
    className="mb-4"
  >
    Add them to a list of your favorite media items.
  </h4>
</div>
`;
